app/src/main/java/de/westnordost/streetcomplete/quests/opening_hours/AddOpeningHours.kt

```diff
@@ -93,12 +93,22 @@ class AddOpeningHours(
             ),
         ).map { it.key + " ~ " + it.value.joinToString("|") }.joinToString("\n or ") + "\n" + """
             )
-            and !opening_hours
+            and (!opening_hours or opening_hours older today -1 years)
+          )
+          or (
+            opening_hours older today -1 years
+            and (
+              leisure = park
+              or barrier
+              or amenity ~ toilets|bicycle_rental
+            )
           )
-          or opening_hours older today -1 years
         )
         and access !~ private|no
-        and (name or brand or noname = yes or name:signed = no or amenity=recycling)
+        and (
+          name or brand or noname = yes or name:signed = no
+          or amenity = recycling or amenity = toilets or barrier
```

Comment:
Match the same logic as above to reduce the cognitive load when looking at the code:
```suggestion
          or barrier
          or amenity ~ toilets|bicycle_rental
```